src/index.ts

```diff
@@ -1,25 +1,475 @@
+      return true;
+    }
+  });
+  return false;
+}
+
+//checkk if the pr contains commit with 'cherry picked from commit xxxx'
+function IsPRContainsCherryPickCommit(pr: any) {
+  listCommitsForPR(client, pr.number).then(response => {
+    response.data.forEach(eachCom => {
+      if (includes(eachCom.commit.message, 'cherry picked from commit')) {
+        //get the branch the commit is from
+        const indexOfSHA = eachCom.commit.message.indexOf('cherry picked from commit');
+        const prefix = 'cherry picked from commit'
+        //length is SHA is 40
+        const SHA = eachCom.commit.message.substring(indexOfSHA + prefix.length - 1 + 2, indexOfSHA + prefix.length - 1 + 42)
+        // supposed to have API to get the branch the commit is from 
+        // and then check if the branch is not master, add comment
+      }
+    })
+  }).catch(logger.error);
+}
+
+function hasFixForIssue(pr: any) {
+  if (pr.hasOwnProperty('body')) {
+    return pr.body.includes("Fixes #")
+  }
+  return false;
+}
+
+//go through all labels, check if there is label with name
+function hasLabelWithName(issue: any, name: String) {
+  const labelList = issue.labels;
+  if (isEmpty(labelList)) {
+    return false;
+  }
+
+  labelList.forEach((label: any) => {
+    if (label.name == name) {
+      return true;
+    }
   });
 
-  return issues;
+  return false;
+}
+
+function hasMoreThanOneTargetLabelThenRemove(issue: any, labelName: string) {
+  const labelsList = issue.labels;
+  let hasTargetLabel = false;
+
+  labelsList.forEach((label: any) => {
+    if (label.name.includes('Target:')) {
+      if (label.name === labelName) {
+        if (hasTargetLabel == false) {
+          hasTargetLabel = true;
+        }
+        else {
+          removeALabelFromIssue(client, issue.number, label.name)
+        }
+      }
+      else {
+        removeALabelFromIssue(client, issue.number, label.name)
+      }
+    }
+  })
 }
+const fetchFreq = userConfig.fetchFreq
+const issueMaxNoStateTime = userConfig.issueMaxNoStateTime
+const issueMaxWaitTimeToRelease = userConfig.issueMaxWaitTimeToRelease
+
+const WEEK_MM = 604800000
+const DAY_MM = 86400000
+const HR_MM = 3600000
+
+function issuesManagement(response: any) {
+  const issuesList = response
+  const times1 = issueMaxNoStateTime.substring(0, issueMaxNoStateTime.indexOf(' '))
+  const unit1 = issueMaxNoStateTime.substring(issueMaxNoStateTime.indexOf(' ') + 1)
+  const times2 = issueMaxWaitTimeToRelease.substring(0, issueMaxNoStateTime.indexOf(' '))
+  const unit2 = issueMaxWaitTimeToRelease.substring(issueMaxNoStateTime.indexOf(' ') + 1)
+
+  issuesList.forEach((issue: any) => {
+
+    // comment on ticket when no 'State:' labels have been added to an issue in 'issueMaxNoStateTime' weeks: userConfig.askForUpdate
+    if (isGreaterThan(Date.now(), new Date(issue.created_at).getTime(), unit1, times1) && !hasLabelWithPrefix(issue, 'State:')) {
+      logger.info("create comment now")
```

Comment:
Log statements like this could include data about the issue/comment so we can better debug in the future. Something like... `logger.info('Commenting request for info on issue #%s', issue.id);`